### PR TITLE
[besu] improve error handling mechanism for vault api calls

### LINKED ISSUE #2260: feat(besu): improve error handling mechanism for vault api calls
**Issue Description:**
The error handling mechanism in the script bevel-vault.sh can be further improved to enhance troubleshooting capabilities and provide better visibility into failures.
This issue aims to enhance the error handling mechanism, improve visibility into failures, and enable effective error handling.

**Current Behavior:**
The script does not have error handling at the individual function level, relying solely on the validateVaultResponseHashicorp function. This approach does not provide granular error handling and may lead to difficulties in identifying and resolving issues.

**Expected Behavior:**
The script should include error-handling mechanisms within each individual function to validate the values set and handle any errors or unexpected conditions encountered. This would ensure that errors are caught and appropriately addressed at each function level, allowing for more effective troubleshooting.

### PR DESCRIPTION
**Pull request by saurabhkumarkardam**
### **Commit to be reviewed**
---

feat(besu): improve error handling mechanism for vault api calls

```
This pull request introduces improvements to the error handling mechanism and enhances troubleshooting capabilities in the bevel-vault.sh script. The changes made include:

 • Implemented error handling at the individual function level.
 • Enhanced error message logging for better visibility into failures.
 • Improved error checking and validation.
 • Updated paths in respect of the function calls to reflect the latest changes.
```

fixes #2260

### CODE REVIEW

File: platforms/hyperledger-besu/charts/node_orion/templates/deployment.yaml [hunk 1]
```diff
@@ -120,16 +120,16 @@ spec:
 
           vault_secret_key="${VAULT_SECRET_ENGINE}/${VAULT_SECRET_PREFIX}/{{ .Values.vault.keyname }}"
           vaultBevelFunc 'readJson' ${vault_secret_key}
-          validateVaultResponseHashicorp "${vault_secret_key}" "${VAULT_SECRET}" "LOOKUPSECRETRESPONSE"
```
Comment (sownak) [inline platforms/hyperledger-besu/charts/node_orion/templates/deployment.yaml:123]: why is this removed? I thought this is what Ravi added

**Review by sownak — CHANGES_REQUESTED** [review]

File: platforms/hyperledger-besu/charts/node_orion/templates/deployment.yaml (hunk 1 above)
Comment (saurabhkumarkardam) [reply platforms/hyperledger-besu/charts/node_orion/templates/deployment.yaml:123]: Yes, just shifted this function within the `bevel-vault.sh` script itself.  Validation will be handled by `bevel-vault.sh` itself, there is no need to define function `validateVaultResponseHashicorp` separately in each different chart.

Comment (suvajit-sarkar) [reply platforms/hyperledger-besu/charts/node_orion/templates/deployment.yaml:123]: the sprint handles error checking and validation for the responses.

**Review by suvajit-sarkar — APPROVED** [review]
LGTM
